var/spack/repos/builtin/packages/amrex/package.py

```diff
@@ -0,0 +1,93 @@
+##############################################################################
+# Copyright (c) 2013-2016, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the LICENSE file for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Amrex(CMakePackage):
+    """A Block-Structured AMR Framework. AMReX is the successor to BoxLib."""
+
+    homepage = "https://ccse.lbl.gov/AMReX/index.html"
+    url      = "https://github.com/AMReX-Codes/amrex.git"
+
+    version('develop', git='https://github.com/AMReX-Codes/amrex.git', tag='master')
+    version('17.06', git='https://github.com/AMReX-Codes/amrex.git', commit='836d3c7')
+
+    variant('mpi', default=True, description='Enable MPI parallel support')
+# variant('omp', default=False, description='Enable OpenMP parallel support')
```

Comment:
> @tgamblin just have some minor philosophical comments w.r.t. how much to assume about the defaults of the package's build system.

Well, I agree. That said, I am unaware of any precedent for this in Spack. I had assumed Spack *should* reflect what package developers themselves seem to think is the best set of defaults. In so doing (not just here but in any package I've worked on), I've coded Spack to tr to only set build variables when they represent non-default behaviors. Is that a bad strategy It probably is as it means the Spack packaging code is then somewhat dependent on whatever defaults the package uses...which could change with time. What is correct approach to take then?